kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/ListAlRichMessage.java

```diff
@@ -0,0 +1,102 @@
+        LinearLayout listItemLayout = containerView.findViewById(R.id.alListMessageLayout);
+        setupAlRichMessage(listItemLayout, model);
+    }
+
+    void setActionTextView(TextView actionTextView, View actionDivider, int index, ALRichMessageModel.ALPayloadModel payload, final List<ALRichMessageModel.AlButtonModel> action, ALRichMessageModel model) {
+        actionTextView.setVisibility(View.VISIBLE);
+        actionTextView.setText(action.get(index).getName());
+        setActionListener(actionTextView, model, action.get(index), payload);
+
+        if(actionDivider != null) {
+            actionDivider.setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    protected void setupAlRichMessage(ViewGroup listItemLayout, ALRichMessageModel model) {
+        super.setupAlRichMessage(listItemLayout, model);
+        if (model != null) {
+            if (model.getPayload() != null) {
+                TextView headerText = listItemLayout.findViewById(R.id.headerText);
+                ImageView headerImage = listItemLayout.findViewById(R.id.headerImage);
+                ALRichMessageModel.ALPayloadModel payload = (ALRichMessageModel.ALPayloadModel) GsonUtils.getObjectFromJson(model.getPayload(), ALRichMessageModel.ALPayloadModel.class);
+                if (payload != null) {
+                    RecyclerView listRecycler = listItemLayout.findViewById(R.id.alListItemRecycler);
+                    LinearLayoutManager layoutManager = new LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false);
+                    listRecycler.setLayoutManager(layoutManager);
+                    AlListRMAdapter adapter = (AlListRMAdapter) AlRichMessageAdapterFactory.getInstance().getListRMAdapter(context, message, payload.getElements(), payload.getReplyMetadata(), listener);
+                    listRecycler.setAdapter(adapter);
+
+                    if (!TextUtils.isEmpty(payload.getHeaderText())) {
+                        headerText.setVisibility(View.VISIBLE);
+                        headerText.setText(getHtmlText(payload.getHeaderText()));
+                    } else {
+                        headerText.setVisibility(View.GONE);
+                    }
+
+                    if (!TextUtils.isEmpty(payload.getHeaderImgSrc())) {
+                        headerImage.setVisibility(View.VISIBLE);
+                        Glide.with(context).load(payload.getHeaderImgSrc()).into(headerImage);
+                    } else {
+                        headerImage.setVisibility(View.GONE);
+                    }
+
+                    if (payload.getButtons() != null) {
+                        final List<ALRichMessageModel.AlButtonModel> action = payload.getButtons();
+
+                        if (action.get(0) != null) {
+                            final TextView actionText1 = listItemLayout.findViewById(R.id.actionButton1);
+                            setActionTextView(actionText1, null, 0, payload, action, model);
```

Comment:
declaring actionText1 is not required, directly use it in next line

Reply:
done